src/coreclr/runtime.proj

```diff
@@ -1,7 +1,26 @@
 <Project Sdk="Microsoft.Build.NoTargets">
 
+  <PropertyGroup>
+    <NativeBuildPartitionPropertiesToRemove>ClrFullNativeBuild;ClrRuntimeSubset;ClrJitSubset;ClrPalTestsSubset;ClrAllJitsSubset;ClrILToolsSubset</NativeBuildPartitionPropertiesToRemove>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="runtime-prereqs.proj" GlobalPropertiesToRemove="$(NativeBuildPartitionPropertiesToRemove)" />
+  </ItemGroup>
+
   <Target Name="BuildRuntime"
-          AfterTargets="Build">
+          BeforeTargets="Build">
+
+    <!-- Get the path to PGO data to pass to the runtime build script. -->
+    <MSBuild
+      Projects=".nuget/optdata/optdata.proj"
+      Targets="GetPgoDataPackagePath">
+      <Output
+        TaskParameter="TargetOutputs"
+        PropertyName="PgoDataPath" />
+    </MSBuild>
```

Comment:
FYI this won't work with static graph build's `-isolate` feature which isn't yet used in the repo but which we might want to leverage in the future.

Reply:
You could avoid this dependency by moving the `GetPgoDataPackagePath` logic into a separate targets files (including the PackageReferences, RuntimeIdentifiers, etc.) and import that in any project that needs access to the calculated path.